File: cpp/easy/2894_divisible_and_non_divisible_sums_difference.cpp
// Posted Solution: https://leetcode.com/problems/divisible-and-non-divisible-sums-difference/solutions/5172788/simple-c-solution-beats-100-of-times/
// https://leetcode.com/problems/divisible-and-non-divisible-sums-difference/description/
class Solution {
public:
    int differenceOfSums(int n, int m){
        int res = 0;
        for(int i = 1; i <= n; i++){
            if((i / m) == (double(i) / double(m))){
                res -= i;
            } else{
                res += i;
            }
        }
        return res;
    }
};

File: python/easy/2894_divisible_and_non_divisible_sums_difference.py
# https://leetcode.com/problems/divisible-and-non-divisible-sums-difference/description/?envType=daily-question&envId=2025-05-27class Solution:
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        res = 0
        for i in range(1, n + 1):
            if i % m == 0:
                res -= i
            else:
                res += i
        return res
